modules/ocf_desktop/files/clean-temp-files

```diff
@@ -1,19 +1,71 @@
-#!/bin/bash -eu
-# Find and remove temporary files (/var/local/tmp/*) from users who haven't
-# logged in recently.
-#
-# For simplicity, /var/local/tmp/ is owned (and only writable) by root. On
-# login, root creates a new directory for the current user to use. We remove
-# these directories periodically via this cron script.
-set -o pipefail
-shopt -s nullglob
-
-cd /var/local/tmp
-for u in *; do
-	# TODO: this check for recent logins is not great; wtmp is rotated monthly
-	# (rather than being a rolling log), so this basically is equivalent to
-	# clearing everyone at the start of the month
-	if ! last -Rw | cut -d' ' -f1 | head -n-2 | grep -q "^$u$"; then
-		rm -rf --one-file-system -- "/var/local/tmp/$u"
-	fi
-done
+#!/usr/bin/env python3
+"""Clean temporary files on desktops if users owning the files have not logged
+in for 2 weeks.
+
+Also makes sure that files are not deleted under a user if they are currently
+logged in to the desktop.
+"""
+import os
+import subprocess
+from datetime import datetime
+from datetime import timedelta
+
+from dateutil import parser
+
+
+def get_command_by_lines(command):
+    """Get the output of a command split by lines"""
+    return subprocess.check_output(command).decode('utf-8').splitlines()
+
+
+def get_current_users():
+    """Get a set of the current users logged in"""
+    who = get_command_by_lines('who')
+
+    return {user.split()[0] for user in who}
+
+
+def get_last_logins():
+    """Get a dictionary of the last time each user logged in to this desktop by
+    parsing the "last" command's output. last uses /var/log/wtmp, which is
+    rotated at the start of each month, so we have to make sure to also check
+    /var/log/wtmp.1 (last month's records) in case it is near the start of the
+    month.
+    """
+    curr_users = get_current_users()
+
+    this_month = get_command_by_lines(('last', '-wF'))
+    last_month = get_command_by_lines(('last', '-wF', '-f', '/var/log/wtmp.1'))
```

Comment:
Hmm, that's a good point, I hadn't considered that, it probably would have some problems, I should check if the file exists first.